diff --git a/pylintrc-google b/pylintrc
index 140a876..0f1c20b 100644
--- a/pylintrc-google
+++ b/pylintrc
@@ -12,10 +12,10 @@ ignore=third_party
 
 # Files or directories matching the regex patterns are skipped. The regex
 # matches against base names, not paths.
-ignore-patterns=
+ignore-patterns=(.)*_test\.py,test_(.)*\.py
 
 # Pickle collected data for later comparisons.
-persistent=no
+persistent=yes
 
 # List of plugins (as comma separated values of python modules names) to load,
 # usually to register additional checkers.
@@ -146,6 +146,8 @@ disable=abstract-method,
         wrong-import-order,
         xrange-builtin,
         zip-builtin-not-iterating,
+        keyword-arg-before-vararg, # This has a specific use case for API methods
+        multiple-statements        # For brevity, we allow multiple statements per line
 
 
 [REPORTS]
@@ -193,7 +195,7 @@ property-classes=abc.abstractproperty,cached_property.cached_property,cached_pro
 function-rgx=^(?:(?P<exempt>setUp|tearDown|setUpModule|tearDownModule)|(?P<camel_case>_?[A-Z][a-zA-Z0-9]*)|(?P<snake_case>_?[a-z][a-z0-9_]*))$
 
 # Regular expression matching correct variable names
-variable-rgx=^[a-z][a-z0-9_]*$
+variable-rgx=^(([a-z][a-z0-9_]*)|([A-Z_]*))$
 
 # Regular expression matching correct constant names
 const-rgx=^(_?[A-Z][A-Z0-9_]*|__[a-z0-9_]+__|_?[a-z][a-z0-9_]*)$
@@ -211,10 +213,10 @@ class-attribute-rgx=^(_?[A-Z][A-Z0-9_]*|__[a-z0-9_]+__|_?[a-z][a-z0-9_]*)$
 inlinevar-rgx=^[a-z][a-z0-9_]*$
 
 # Regular expression matching correct class names
-class-rgx=^_?[A-Z][a-zA-Z0-9]*$
+class-rgx=^__|_?[A-Z][a-zA-Z0-9]*$
 
 # Regular expression matching correct module names
-module-rgx=^(_?[a-z][a-z0-9_]*|__init__)$
+module-rgx=^(__|_?[a-z][a-z0-9_]*|__init__|__main__)$
 
 # Regular expression matching correct method names
 method-rgx=(?x)^(?:(?P<exempt>_[a-z0-9_]+__|runTest|setUp|tearDown|setUpTestCase|tearDownTestCase|setupSelf|tearDownClass|setUpClass|(test|assert)_*[A-Z0-9][a-zA-Z0-9_]*|next)|(?P<camel_case>_{0,2}[A-Z][a-zA-Z0-9_]*)|(?P<snake_case>_{0,2}[a-z][a-z0-9_]*))$
@@ -259,14 +261,14 @@ generated-members=
 [FORMAT]
 
 # Maximum number of characters on a single line.
-max-line-length=80
+max-line-length=88
 
 # TODO(https://github.com/PyCQA/pylint/issues/3352): Direct pylint to exempt
 # lines made too long by directives to pytype.
 
 # Regexp for a line that is allowed to be longer than the limit.
 ignore-long-lines=(?x)(
-  ^\s*(\#\ )?<?https?://\S+>?$|
+  ^\s*(\#\ )?(@see\ )?<?https?://\S+>?$|
   ^\s*(from\s+\S+\s+)?import\s+.+$)
 
 # Allow the body of an if to be on the same line as the test if there is no
@@ -299,7 +301,7 @@ notes=TODO
 
 # This flag controls whether inconsistent-quotes generates a warning when the
 # character used as a quote delimiter is used inconsistently within a module.
-check-quote-consistency=yes
+check-quote-consistency=no
 
 
 [VARIABLES]
@@ -427,3 +429,10 @@ valid-metaclass-classmethod-first-arg=mcs
 overgeneral-exceptions=builtins.StandardError,
                        builtins.Exception,
                        builtins.BaseException
+
+; [DOCSTRING]
+
+; default-docstring-type=google
+; # List of exceptions that do not need to be mentioned in the Raises section of
+; # a docstring.
+; ignore-exceptions=AssertionError,NotImplementedError,StopIteration,TypeError
